### PR TITLE
use spherical_harmonics from scipy in orbital function

### LINKED ISSUE #1839: Add test for orbital
### Describe the Issue!

The function `orbital` is not tested and there seems to have an error in the spherical harmonics computation.
Adding a test and fixing any error, possibly using `scipy.special.sph_harm`, would be helpful.

### PR DESCRIPTION
**Pull request by christian512**
**Description**
The orbital function had some errors in it. Here, I replaced the manual calculation of the spherical harmonics, with the implementation provided by SciPy. Additionally, I provide some tests to the function.

**Related issues or PRs**
fixes #1839 

**Changelog**
Fixed errors in orbital functions

### CODE REVIEW

File: qutip/tests/test_orbital.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    phi_list = np.linspace(0, 2 * np.pi, num=100)
+    l, m = 5, 2
+    q1 = qutip.basis(2 * l + 1, l + m)
+    l, m = 3, -1
+    q2 = qutip.basis(2 * l + 1, l + m)
+    for theta, phi in zip(theta_list, phi_list):
+        assert orbital(theta, phi, q1, q2) == sph_harm(2, 5, phi,
+                                                       theta) + sph_harm(-1, 3,
+                                                                         phi,
+                                                                         theta)
```
Comment (Ericgig) [inline qutip/tests/test_orbital.py:35]: This does not look clean.
Computing `expected` on another line would look better.

**Review by Ericgig — COMMENTED** [review]
Hi christian512,
Thank you for the PR.
I let some comments where you can simplify the code, but overall, it looks great.

File: qutip/orbital.py [hunk 2]
```diff
@@ -11,94 +11,77 @@ def orbital(theta, phi, *args):
...
 
+    # Convert theta and phi to numpy np arrays and check dimensions
+    if isinstance(theta, float) or isinstance(theta, int):
+        theta = np.array([theta])
+    if isinstance(phi, float) or isinstance(phi, int):
+        phi = np.array([phi])
+    if isinstance(theta, list):
+        theta = np.array(theta)
+    if isinstance(phi, list):
+        phi = np.array(phi)
```
[the remark this answers is not in the document]
Comment (christian512) [reply qutip/orbital.py:47]: That is true. But if a float or int is provided, `np.asarray` does not give the desired list form. So I think I have to keep the two checks for float and int.

[the remark this answers is not in the document]
Comment (Ericgig) [reply qutip/orbital.py]: `np.atleast_1d`?

**Review by Ericgig — APPROVED** [review]
Look good.

File: qutip/orbital.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 __all__ = ['orbital']
 
 import numpy as np
-from scipy.special import factorial
+from scipy.special import factorial, sph_harm
```
Comment (Ericgig) [inline qutip/orbital.py:4]: `factorial` is no longer used.
```suggestion
from scipy.special import sph_harm
```

**Comment by hodgestar** [comment]
Doh, I merged this before the unused import was removed.

@christian512 Could you make a tiny PR to remove the unused import? Apologies for not committing it myself before merging.

**Comment by christian512** [comment]
@hodgestar: Created the tiny PR here: https://github.com/qutip/qutip/pull/1902

**Comment by hodgestar** [comment]
@christian512 Thanks! I merged #1902.

**Pull request merged**